src/xcode/ENA/ENA/Source/Services/Risk/Provider/Helper/DeviceTimeCheck.swift

```diff
@@ -24,50 +23,53 @@ final class DeviceTimeCheck: DeviceTimeCheckProtocol {
 
 	// MARK: - Internal
 
-	var isDeviceTimeCorrect: Bool {
-		store.isDeviceTimeCorrect
+	enum TimeCheckResult: Int, Codable {
+		case correct
+		case assumedCorrect
+		case incorrect
 	}
 
-	var wasDeviceTimeErrorShown: Bool {
-		store.isDeviceTimeCorrect
+	var isDeviceTimeCorrect: Bool {
+		return store.deviceTimeCheckResult == .correct || store.deviceTimeCheckResult == .assumedCorrect
 	}
 
-	func updateDeviceTimeFlags(serverTime: Date, deviceTime: Date) {
-		self.persistDeviceTimeCheckFlags(
-			isDeviceTimeCorrect: self.isDeviceTimeCorrect(
-				serverTime: serverTime,
-				deviceTime: deviceTime
-			),
-			isDeviceTimeCheckKillSwitchActive: self.isDeviceTimeCheckKillSwitchActive(
-				config: self.store.appConfigMetadata?.appConfig
-			)
+	func updateDeviceTimeFlags(serverTime: Date, deviceTime: Date, configUpdateSuccessful: Bool) {
+		let oldState = store.deviceTimeCheckResult
+		store.deviceTimeCheckResult = isDeviceTimeCorrect(
+			serverTime: serverTime,
+			deviceTime: deviceTime,
+			configUpdateSuccessful: configUpdateSuccessful
 		)
+		// store change date only if a state change was detected
+		if oldState != store.deviceTimeCheckResult {
+			store.deviceTimeLastStateChange = Date()
+		}
+
+        if store.deviceTimeCheckResult == .correct {
+            store.wasDeviceTimeErrorShown = false
+        }
 	}
 
-	func resetDeviceTimeFlags() {
-		store.isDeviceTimeCorrect = true
+	func resetDeviceTimeFlags(configUpdateSuccessful: Bool) {
+		store.deviceTimeCheckResult = configUpdateSuccessful ? .correct : .assumedCorrect
```

Comment:
I'm not sure about setting to correct if the date header was missing in the app config request.
I also can't find anything about that in the spec.

Reply:
good point, I'll ask Max about that - I was a bit confuse about that fact as well

Reply:
Max said we should set it to assume_correct - I will change it.